### PR TITLE
Add a report for invalid location uris

### LINKED ISSUE #4152: Report on location URIs in descriptive metadata
Criteria: values of `event.location.uri` that do not begin with `http://` or `https://` or which end with `.html`.

Output: object druid, object catkey, collection druid, value of `uri`, value of sibling `code`, value of sibling `value`

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔
Fixes #4152

## How was this change tested? 🤨
report here https://gist.github.com/jcoyne/c671e5223885cd9fcb702be7ff313cf2

### CODE REVIEW

File: app/reports/invalid_location_uris.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Invoke via:
+# bin/rails r -e production "InvalidLocationUris.report"
+class InvalidLocationUris
+  JSON_PATH = '$.**.event.location.uri'
+  SQL = <<~SQL.squish.freeze.squish
```
Comment (mjgiarlo) [inline app/reports/invalid_location_uris.rb:7]: yo dawg i heard u like squish?

File: app/reports/invalid_location_uris.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    SELECT (jsonb_path_query_array(description, '#{JSON_PATH} ? (@ like_regex "^.*\.html$")') ||
+            jsonb_path_query_array(description, '#{JSON_PATH} ? (@ like_regex "^(?!https?://).*$")')) ->> 0 as contrib,
```
Comment (mjgiarlo) [inline app/reports/invalid_location_uris.rb:9]: How do you feel about updating this binding name relative to what's in the JSON PATH above?

File: app/reports/invalid_location_uris.rb (hunk 1 above)
Comment (jcoyne) [reply app/reports/invalid_location_uris.rb:7]: Rubocop does this. I agree it seems dumb.

File: app/reports/invalid_location_uris.rb (hunk 2 above)
Comment (jcoyne) [reply app/reports/invalid_location_uris.rb]: updated to the generic `value`

File: app/reports/invalid_location_uris.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/reports/invalid_location_uris.rb:7]: @jcoyne Odd. Which cop? Seems strange to squish an already squished frozen string.

Comment (jcoyne) [reply app/reports/invalid_location_uris.rb:7]: If you run with `-A` this is what you get 🤷

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**